inc/healthcheck/cavalcade/namespace.php

```diff
@@ -51,10 +51,13 @@ function set_last_run() {
  * @return boolean|WP_Error True if healthy, error otherwise.
  */
 function check_health() {
+	switch_to_blog( get_network()->id );
```

Comment:
This get's the network's ID, not the main blog on the network. In WordPress you can have many networks, with many blogs in each network. Confusingly "site" can either mean "blog" or "network" depending on when the function was written.

We only enqueue this job when `is_main_site()` is `true`, `is_main_site` uses `get_main_site_id()` (site in this context means "blog"), so this should use `switch_to_blog( get_main_site_id() );`

We do have another bug here though, because if the site has many networks (which isn't default with WordPress, but is possible to set up), we'll be registering one job per network (on the main site of each network). I don't thikn we need to fix that in this PR, but we should make a ticket, cc @rmccue who wrote the intial cavalcade job checker, does that make sense?

Reply:
Good point - I considered `get_main_site_id()` but wasn't sure if the network ID is not better - obviously the confusion comes from the way these are called. Just changed it.

Reply:
@joehoyle Yeah, I think we probably want to switch to the main blog of the main network, which would be `switch_to_blog( get_network( get_main_network_id() )->site_id )`

Reply:
@rmccue @joehoyle so finally which one is correct?

Reply:
What you changed it to is good! We’ll want to fix that other issue at some point but doesn’t need to happen in this pr